contracts/defi/provider/MakerProvider.sol

```diff
@@ -0,0 +1,818 @@
+ * that may be used to orchestrate the exchange of tokens.
+ * @author Olivier VDB - <olivier@argent.xyz>, Julien Niset - <julien@argent.xyz>
+ */
+contract MakerProvider is Loan { //}, Leverage {
+
+    // Mock token address for ETH
+    address constant internal ETH_TOKEN_ADDRESS = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+
+    // Method signatures to reduce gas cost at depoyment
+    bytes4 constant internal CDP_DRAW = bytes4(keccak256("draw(bytes32,uint256)"));
+    bytes4 constant internal CDP_WIPE = bytes4(keccak256("wipe(bytes32,uint256)"));
+    bytes4 constant internal CDP_SHUT = bytes4(keccak256("shut(bytes32)"));
+    bytes4 constant internal CDP_JOIN = bytes4(keccak256("join(uint256)"));
+    bytes4 constant internal CDP_LOCK = bytes4(keccak256("lock(bytes32,uint256)"));
+    bytes4 constant internal CDP_FREE = bytes4(keccak256("free(bytes32,uint256)"));
+    bytes4 constant internal CDP_EXIT = bytes4(keccak256("exit(uint256)"));
+    bytes4 constant internal WETH_DEPOSIT = bytes4(keccak256("deposit"));
+    bytes4 constant internal WETH_WITHDRAW = bytes4(keccak256("withdraw(uint256)"));
+    bytes4 constant internal ERC20_APPROVE = bytes4(keccak256("approve(address,uint256)"));
+    bytes4 constant internal ETH_TOKEN_SWAP_OUTPUT = bytes4(keccak256("ethToTokenSwapOutput(uint256,uint256)"));
+    bytes4 constant internal ETH_TOKEN_SWAP_INPUT = bytes4(keccak256("ethToTokenSwapInput(uint256,uint256)"));
+    bytes4 constant internal TOKEN_ETH_SWAP_INPUT = bytes4(keccak256("tokenToEthSwapInput(uint256,uint256,uint256)"));
+
+    using SafeMath for uint256;
+
+    /* *************** Events *************************** */
+
+    event CdpOpened(address indexed wallet, bytes32 cup, uint256 pethCollateral, uint256 daiDebt);    
+    event CdpUpdated(address indexed wallet, bytes32 cup, uint256 pethCollateral, uint256 daiDebt);    
+    event CdpClosed(address indexed wallet, bytes32 cup);
+
+    /* ********************************** Implementation of Loan ************************************* */
+
+   /**
+     * @dev Opens a collateralized loan.
+     * @param _wallet The target wallet.
+     * @param _collateral The token used as a collateral (must be 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE).
+     * @param _collateralAmount The amount of collateral token provided.
+     * @param _debtToken The token borrowed (must be the address of the DAI contract).
+     * @param _debtAmount The amount of tokens borrowed.
+     * @param _oracles An array containing the address of MakerDAO's Tub contract as first element.
+     * @return The ID of the created CDP.
+     */
+    function openLoan(
+        BaseWallet _wallet, 
+        address _collateral, 
+        uint256 _collateralAmount, 
+        address _debtToken, 
+        uint256 _debtAmount, 
+        address[] calldata _oracles
+    ) 
+        external 
+        returns (bytes32 _loanId)
+    {
```

Comment:
Would be nice to add `require(_oracles.length == 1)` as you did for the other providers